### ISSUE TITLE
OpenJ9 cannot build android project.

### ISSUE THREAD

**Issue by abcfy2** [AdoptOpenJDK/openjdk-support#127]
* **Platform and architecture (e.g. Linux/x64**:

Linux/x64

* **Java version (from "java --version") e.g. 1.8.0_181, 11.0.4+11 etc.**:

openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20200416_608 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - cfa47e66cd5 based on jdk8u252-b09)

* **Variant e.g. HotSpot or OpenJ9**:

OpenJ9

* **Have you tested it with other vendors' java**:

Yes.

* **Steps to reproduce**:

I just follow the Android guide to create a new project: https://developer.android.com/training/basics/firstapp/creating-project

Here is the sample:

[archive.zip](https://github.com/AdoptOpenJDK/openjdk-support/files/4772444/archive.zip)

But build failed.

```shell
$ java -version
openjdk version "1.8.0_252"
OpenJDK Runtime Environment (build 1.8.0_252-b09)
Eclipse OpenJ9 VM (build openj9-0.20.0, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20200416_608 (JIT enabled, AOT enabled)
OpenJ9   - 05fa2d361
OMR      - d4365f371
JCL      - cfa47e66cd5 based on jdk8u252-b09)

# Clean all gradle caches
$ pkill java
$ rm -fr ~/.gradle/caches ~/.gradle/daemon ~/.gradle/kotlin-profile ~/.gradle/native ~/.gradle/notifications ~/.gradle/workers

$ ./gradlew assemble
# build failed
```

* **Error messsage and/or description of what is not working as expected**:

Here is the output
```

Welcome to Gradle 5.6.4!

Here are the highlights of this release:
 - Incremental Groovy compilation
 - Groovy compile avoidance
 - Test fixtures for Java projects
 - Manage plugin versions via settings script

For more details see https://docs.gradle.org/5.6.4/release-notes.html

To honour the JVM settings for this build a new JVM will be forked. Please consider using the daemon: https://docs.gradle.org/5.6.4/userguide/gradle_daemon.html.
Daemon will be stopped at the end of the build stopping after processing
> Task :clean UP-TO-DATE
> Task :app:clean
> Task :app:preBuild UP-TO-DATE
> Task :app:preDebugBuild UP-TO-DATE
> Task :app:compileDebugAidl NO-SOURCE
> Task :app:generateDebugBuildConfig
> Task :app:compileDebugRenderscript NO-SOURCE
> Task :app:writeDebugApplicationId
> Task :app:generateSafeArgsDebug
> Task :app:javaPreCompileDebug
> Task :app:mainApkListPersistenceDebug
> Task :app:generateDebugResValues
> Task :app:generateDebugResources
> Task :app:createDebugCompatibleScreenManifests
> Task :app:extractDeepLinksDebug
> Task :app:processDebugManifest
> Task :app:mergeDebugResources
> Task :app:processDebugResources

> Task :app:compileDebugJavaWithJavac
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/FirstFragmentDirections.java:30: warning: [unchecked] unchecked call to put(K,V) as a member of the raw type HashMap
      this.arguments.put("myArg", myArg);
                        ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/FirstFragmentDirections.java:38: warning: [unchecked] unchecked call to put(K,V) as a member of the raw type HashMap
      this.arguments.put("myArg", myArg);
                        ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/SecondFragmentArgs.java:20: warning: [unchecked] unchecked call to putAll(Map<? extends K,? extends V>) as a member of the raw type HashMap
    this.arguments.putAll(argumentsMap);
                         ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/SecondFragmentArgs.java:94: warning: [unchecked] unchecked call to putAll(Map<? extends K,? extends V>) as a member of the raw type HashMap
      this.arguments.putAll(original.arguments);
                           ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/SecondFragmentArgs.java:101: warning: [unchecked] unchecked call to put(K,V) as a member of the raw type HashMap
      this.arguments.put("myArg", myArg);
                        ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
/home/fengyu/projects/MyAndroidApp/app/build/generated/source/navigation-args/debug/com/example/myandroidapp/SecondFragmentArgs.java:115: warning: [unchecked] unchecked call to put(K,V) as a member of the raw type HashMap
      this.arguments.put("myArg", myArg);
                        ^
  where K,V are type-variables:
    K extends Object declared in class HashMap
    V extends Object declared in class HashMap
6 warnings

> Task :app:compileDebugSources
> Task :app:mergeDebugShaders
> Task :app:compileDebugShaders
> Task :app:generateDebugAssets
> Task :app:mergeDebugAssets
> Task :app:processDebugJavaRes NO-SOURCE
> Task :app:dexBuilderDebug
> Task :app:checkDebugDuplicateClasses
> Task :app:mergeDebugJavaResource
> Task :app:mergeLibDexDebug
> Task :app:mergeDebugJniLibFolders
> Task :app:validateSigningDebug
> Task :app:preReleaseBuild UP-TO-DATE
> Task :app:compileReleaseAidl NO-SOURCE
> Task :app:compileReleaseRenderscript NO-SOURCE
> Task :app:generateReleaseBuildConfig
> Task :app:writeReleaseApplicationId
> Task :app:generateSafeArgsRelease
> Task :app:javaPreCompileRelease
> Task :app:mainApkListPersistenceRelease
> Task :app:mergeDebugNativeLibs
> Task :app:stripDebugDebugSymbols
> Task :app:generateReleaseResValues
> Task :app:generateReleaseResources
> Task :app:createReleaseCompatibleScreenManifests
> Task :app:extractDeepLinksRelease
> Task :app:mergeProjectDexDebug
> Task :app:processReleaseManifest

> Task :app:mergeExtDexDebug FAILED
java.lang.IllegalArgumentException: java.lang.IllegalArgumentException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
        at java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:593)
        at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1005)
        at com.android.tools.r8.utils.ThreadUtils.a(:4)
        at com.android.tools.r8.dex.a.a(:70)
        at com.android.tools.r8.dex.a.a(:12)
        at com.android.tools.r8.D8.d(:6)
        at com.android.tools.r8.D8.b(:1)
        at com.android.tools.r8.D8$$Lambda$584/0000000000000000.run(Unknown Source)
        at com.android.tools.r8.utils.O.a(:30)
        at com.android.tools.r8.D8.run(:11)
        at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:117)
        at com.android.build.gradle.internal.transforms.DexMergerTransformCallable.call(DexMergerTransformCallable.java:102)
        at com.android.build.gradle.internal.tasks.DexMergingTaskRunnable.run(DexMergingTask.kt:445)
        at com.android.build.gradle.internal.tasks.Workers$ActionFacade.run(Workers.kt:348)
        at org.gradle.workers.internal.AdapterWorkAction.execute(AdapterWorkAction.java:50)
        at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:47)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1$1.create(NoIsolationWorkerFactory.java:65)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1$1.create(NoIsolationWorkerFactory.java:61)
        at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.execute(NoIsolationWorkerFactory.java:61)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
        at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
        at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
        at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
        at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
        at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:56)
        at org.gradle.workers.internal.DefaultWorkerExecutor$3.call(DefaultWorkerExecutor.java:215)
        at org.gradle.workers.internal.DefaultWorkerExecutor$3.call(DefaultWorkerExecutor.java:210)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
        at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
        at java.lang.Thread.run(Thread.java:821)
        Suppressed: java.util.concurrent.ExecutionException: java.lang.IllegalArgumentException: java.lang.IllegalArgumentException
                at java.util.concurrent.ForkJoinTask.get(ForkJoinTask.java:1006)
                ... 43 more
        [CIRCULAR REFERENCE:java.lang.IllegalArgumentException: java.lang.IllegalArgumentException]
Caused by: java.lang.IllegalArgumentException
        at java.nio.Buffer.position(Buffer.java:244)
        at com.android.tools.r8.dex.g.c(:3)
        at com.android.tools.r8.dex.BinaryReader.position(:1)
        at com.android.tools.r8.dex.j.a(:240)
        at com.android.tools.r8.dex.a$a.a(:3)
        at com.android.tools.r8.dex.a$a$$Lambda$775/0000000000000000.run(Unknown Source)
        at java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1386)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)


> Task :app:mergeReleaseResources

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeExtDexDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > java.lang.IllegalArgumentException: java.lang.IllegalArgumentException

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 17s
37 actionable tasks: 36 executed, 1 up-to-date
```

But HotSpot is OK.

**Comment by aahlenst** [comment]
@pshipton Could you have a look, please? The problem might actually be in Android's code shrinker (R8) or desugaring engine (D8) that are being executed. Maybe they depend on some HotSpot specifics.

**Comment by pshipton** [comment]
I've created an issue at the OpenJ9 project to investigate the problem. https://github.com/eclipse/openj9/issues/9871

**Comment by DanHeidinga** [comment]
Can you retest with the latest 0.21.0 builds?  I expect the fix for eclipse/openj9#8823 should resolve this as well - see update in https://github.com/eclipse/openj9/issues/9871#issuecomment-653653164

**Comment by abcfy2** [comment]
Great. Seem that this issue has been fixed. Thanks.
